src/main/java/com/khomishchak/goalsservice/config/GoalServiceConfig.java

```diff
@@ -0,0 +1,14 @@
+package com.khomishchak.goalsservice.config;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.client.RestTemplate;
+
+@Configuration
+public class GoalServiceConfig {
+
+    @Bean("goalServiceRestTemplate")
+    public RestTemplate getRestTemplate() {
```

Comment:
Why do you need to qualify it ?

Reply:
isn't it better to always qualify beans in order to avoid any potential mismatches?